api/src/modules/materials/materials.service.ts

```diff
@@ -170,25 +170,29 @@ export class MaterialsService extends AppBaseService<
 
   async getMaterialsTreeWithSourcingLocations(
     materialTreeOptions: GetMaterialTreeWithOptionsDto,
+    descendantsFound?: boolean,
   ): Promise<Material[]> {
-    if (materialTreeOptions.originIds) {
-      materialTreeOptions.originIds =
-        await this.adminRegionService.getAdminRegionDescendants(
-          materialTreeOptions.originIds,
-        );
-    }
-    if (materialTreeOptions.businessUnitIds) {
-      materialTreeOptions.businessUnitIds =
-        await this.businessUnitsService.getBusinessUnitsDescendants(
-          materialTreeOptions.businessUnitIds,
-        );
-    }
-    if (materialTreeOptions.supplierIds) {
-      materialTreeOptions.supplierIds =
-        await this.suppliersService.getSuppliersDescendants(
-          materialTreeOptions.supplierIds,
-        );
+    if (!descendantsFound) {
+      if (materialTreeOptions.originIds) {
+        materialTreeOptions.originIds =
+          await this.adminRegionService.getAdminRegionDescendants(
+            materialTreeOptions.originIds,
+          );
+      }
+      if (materialTreeOptions.businessUnitIds) {
+        materialTreeOptions.businessUnitIds =
+          await this.businessUnitsService.getBusinessUnitsDescendants(
+            materialTreeOptions.businessUnitIds,
+          );
+      }
+      if (materialTreeOptions.supplierIds) {
+        materialTreeOptions.supplierIds =
+          await this.suppliersService.getSuppliersDescendants(
+            materialTreeOptions.supplierIds,
+          );
+      }
     }
+
```

Comment:
In order to avoid having to decide based on an external context of where the method is called (do the options have descendants ids or not), it would probably be better to move this responsibility to the caller, and, let the function do the processing on what it gets, and not make it more ambiguous based on an external context. This applies to the same variants of the function on each service (Materials, suppliers, etc)